config.schema.yaml

```diff
@@ -33,28 +33,40 @@ enterprise_url:
   type: "string"
 
 deployment_environment:
-  description: "The enviroment for this StackStorm server."
+  description: "The environment for this StackStorm server."
   type: "string"
   required: true
   default: "production"
 
-# You need to manually put this into your config, as it's not
-# compatible with the config schema.
-
-#repository_sensor:
-#  event_type_whitelist:
-#    - "IssuesEvent"
-#    - "IssueCommentEvent"
-#    - "ForkEvent"
-#    - "WatchEvent"
-#    - "ReleaseEvent"
-#    - "PushEvent"
-#
-#  repositories:
-#    -
-#      user: "StackStorm"
-#      name: "st2"
-#    -
-#      user: "StackStorm"
-#      name: "st2contrib"
-#  count: 30  # Maximum number of old events to retrieve
+repository_sensor:
+  description: "Sensor specific settings."
+  type: "object"
+  required: false
+  additionalProperties: false
+  properties:
+    event_type_whitelist:
+      type: "array"
+      description: "Event types to watch for, e.g. IssuesEvent, PushEvent"
```

Comment:
To clarify - this is a "required" value for sensor because if you don't specify any event types here the sensor will simply ignore all the events and not do any work.

So other that or make it a required attribute :)